.cursor/rules/submission-contracts.mdc

```diff
@@ -0,0 +1,169 @@
+---
+description: 
+globs: 
+alwaysApply: true
+---
+---
+description: Guidelines for implementing submission contracts in WAVS components
+globs: src/contracts/**/*.sol
+---
+# Submission Contract Guidelines
+
+Guidelines for implementing submission contracts that handle WAVS component outputs.
+
+## Basic Structure
+
+1. Required Interfaces:
+```solidity
+// Required imports
+import {IWavsServiceManager} from "@wavs/interfaces/IWavsServiceManager.sol";
+import {IWavsServiceHandler} from "@wavs/interfaces/IWavsServiceHandler.sol";
+
+// Contract declaration
+contract YourSubmitContract is IWavsServiceHandler {
+    // Service manager instance
+    IWavsServiceManager private _serviceManager;
+
+    constructor(IWavsServiceManager serviceManager) {
+        _serviceManager = serviceManager;
+    }
+}
+```
+
+2. Required State Variables:
+```solidity
+// Mapping for tracking valid triggers
+mapping(TriggerId _triggerId => bool _isValid) internal _validTriggers;
+
+// Mapping for storing trigger data
+mapping(TriggerId _triggerId => bytes _data) internal _datas;
+
+// Mapping for storing operator signatures
+mapping(TriggerId _triggerId => bytes _signature) internal _signatures;
+```
```

Comment:
These are NOT required.

Reply:
What would be better is a discussion about TriggerIds and the role they play... not every service needs a triggerId.